### PR TITLE
Upgrade to the Firebase 3.x.x SDK

### PR DESCRIPTION
**Pull request by lmckenzie**
I tool a stab at updating geofire to use firebase SDK version 3. It is probably not perfect, I had trouble getting all the tests to work and I will need to revisit after I know what the FirebaseOptions for testing should be.

### CODE REVIEW

**Comment by lmckenzie** [comment]
I signed it

**Comment by abehaskins** [comment]
This is great - I'll take a gander soon and see how the tests look and see what we can do about cleaning this up and getting it merged in.

File: src/main/java/com/firebase/geofire/GeoQuery.java [hunk 1]
```diff
@@ -112,8 +115,8 @@ private boolean locationIsInQuery(GeoLocation location) {
     }
 
     private void postEvent(Runnable r) {
-        EventTarget target = Firebase.getDefaultConfig().getEventTarget();
-        target.postEvent(r);
+        Repo repo = geoFire.getDatabaseReference().getRepo();
```
Comment (abehaskins) [inline src/main/java/com/firebase/geofire/GeoQuery.java]: Can you elaborate on your choice to use this internal tool for the runnable?

Comment (lmckenzie) [inline src/main/java/com/firebase/geofire/GeoQuery.java]: I could not find a way to access the EventTarget similar to the existing code. After decompiling the library and looking around, this looked like the closest replication of the existing behavior. If you  have a recommendation regarding how I should have done this I can update the PR.

**Comment by lmckenzie** [comment]
@jdr1 thanks! I was not aware of that api change. I have updated the PR.

File: src/main/java/com/firebase/geofire/GeoQuery.java (hunk 1 above)
Comment (HammadNasir) [inline src/main/java/com/firebase/geofire/GeoQuery.java]: How can I update GeoFire.java file in my app?

**Comment by lmckenzie** [comment]
I have updated the PR.

After implementing this fix, adjusting the RealDataTest setup, and using a valid account with the tests I have them all passing.

**Comment by framuroni** [comment]
Hi! i'm trying to use this updated version, but i cannot compile it as it does not find EventTarget and Repo, can you help me?

**Comment by jdimond** [comment]
Thanks for doing this, merging. We'll need to do a bit more clean up until we can release a new version.

**Pull request merged**